### DISCUSSION TITLE
Builds for sub-assemblies

### DISCUSSION THREAD

**Discussion by camiller000** [inventree/InvenTree#4201] [Q&A]
We have a small manufacturing team at the moment and most products built in small batches of 25.  We are beginning to use Inventree to help us prepare to scale.

Our first use case has a product that has a dozen subassemblies; some that go 3 to 4 levels deep.  For the time being, we do not foresee that we would be building any of those subassemblies separate and apart from the main build.  Doing a build for each child level is time consuming.  But without doing a build for each we can't allocate stock for those parts.

Is there a way to batch create all child builds required?  I see on the BOM report you can control the levels reported (e.g. 0 = all levels).

Any insights appreciated.

**Comment by matmair** [comment]
This is not a function we currently have built in, though I see nothing obvious that would stop an addition to support this.
Maybe file a feature request and we see if someone is interested in developing it.

**Reply by camiller000** [reply]
OK I will do that.

**Comment by miggland** [comment]
Without having checked each step, this feels like something which could be achieved using the API and some tool such as `inventree-python`, without the need to change something in core (you'd probably have your implementation of exactly what you want faster this way..)

**Reply by matmair** [reply]
@miggland I would prefer providing solutions that do not involve creating one-offs. It can be very intmimtating for users to have to code something up and most - if not all - scripts end up not being shared, thus increasing the effort for everyone.

**Reply by miggland** [reply]
Makes sense, as long as you judge the solution to be applicable in general. However, to be fair to the users I still think it's worth pointing out other possible solutions, which can be implemented soon, and don't require being put in a (rather long) list of feature requests..

**Reply by camiller000** [reply]
I appreciate the alternative recommendation.  We do not have the expertise in-house to take this on but would be willing to engage a resource that knows the platform that could do this for us.

**Comment by SchrodingersGat** [comment]
Ref: https://github.com/inventree/InvenTree/issues/4212
